lib/src/models/view/schedule_search.dart

```diff
@@ -18,11 +18,14 @@ class ScheduleSearchModel {
 
 	/// Finds all courses that match the given query.
 	/// 
-	/// The query parameter should be a part of a course's name, and classes will
-	/// be matched by searching against [Subject.name].
+	/// The query parameter should be a part of a course's name, id, or teacher
+	/// and classes will be matched by searching against
+	/// [Subject.name], [Subject.id], or [Subject.teacher].
```

Comment:
```suggestion
	/// and classes will be matched by searching against
	/// [Subject.name], [Subject.id], or [Subject.teacher].
	///
	/// [query] should be in lower-case to allow broader matching.
```

If we force the UI to provide lowercase values, then `query.toLowerCase()` is run only once (when the UI passes it to your model) instead of multiple times (in each `||`). It's technically more efficient but more importantly it makes the code look better. 

Or, you can add a line `query = query.toLowerCase();` before returning the list, so that way it works no matter what the UI passes through.